### ISSUE TITLE
GPU double sided, reverse face issue?

### ISSUE THREAD

**Issue by Dentzz1** [ChaosGroup/vray-for-houdini#282]
Using the GPU render I am having issues with what I think is double sided or reversed faces. I created a box deleted one of the sides and rendered. Vray CPU is able to correctly render this out if double sided is checked. 
![cpubox](https://user-images.githubusercontent.com/22080153/32997755-2dfcf60c-cd62-11e7-90cb-0236f18fc0bc.jpg)

The GPU does not correctly render this out and gets the same results as the cpu does with double sided unchecked.
![gpubox](https://user-images.githubusercontent.com/22080153/32997756-31df027e-cd62-11e7-8332-a11a144e4b4c.jpg)

If I reverse the faces it is able to render out a single side but reverse face still gives strange results.
![gpuboxreverse](https://user-images.githubusercontent.com/22080153/32997786-74a72abe-cd62-11e7-8699-e9ab194fe0a7.jpg)

Vray GPU I think should be able to handle these backface polygons.

**Comment by bravehamster** [comment]
Hey there! 
There are a couple of things going on here.
First of all, yes, it seems that the GPU won't render backfaces even if DoubleSided is enabled.

The faces on your second screenshot are shaded for another reason - when 2-sided is disabled on the V-Ray material and you're looking at the back side of a face that is receiving illumination from a light source, it seems that the shading of the front side of the face is seen through the back. 

So in a nutshell - the GPU is shading your geometry as if its single-sided and you're seeing the shading from the front side of the polygons. 

Thanks for reporting that, good catch !
